File: backend/sallesMangement/src/main/java/com/projetee/sallesmangement/service/CategoryService.java
package com.projetee.sallesmangement.service;

import com.projetee.sallesmangement.dto.category.CategoryRequest;
import com.projetee.sallesmangement.dto.category.CategoryResponse;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CategoryService {

    CategoryResponse create(CategoryRequest request);

    CategoryResponse get(Long id);

    List<CategoryResponse> getAll();

    Page<CategoryResponse> getPaginated(int page, int size, String sortBy);

    CategoryResponse update(Long id, CategoryRequest request);

    void delete(Long id);
}

File: frontend/sales-management-frontend/src/services/categoryService.js
import api from './api';

export const categoryService = {
  async getAll() {
    const response = await api.get('/categories');
    return response.data;
  },

  async getPaginated(page = 0, size = 10, sortBy = 'name') {
    const response = await api.get('/categories/page', {
      params: { page, size, sortBy }
    });
    return response.data;
  },

  async getById(id) {
    const response = await api.get(`/categories/${id}`);
    return response.data;
  },

  async create(data) {
    const response = await api.post('/categories', data);
    return response.data;
  },

  async update(id, data) {
    const response = await api.put(`/categories/${id}`, data);
    return response.data;
  },

  async delete(id) {
    await api.delete(`/categories/${id}`);
  }
};